server/camcops_server/templates/snippets/describe_task_filter.mako

```diff
@@ -43,65 +43,69 @@ from camcops_server.cc_modules.cc_constants import DateFormat
 <div class="filters">
     ## Task types
     %if task_filter.task_types:
-        ${_("Task is one of:")} <b>${ ", ".join(task_filter.task_types) | h }</b>.
+        ${ _("Task is one of:") } <b>${ ", ".join(task_filter.task_types) }</b>.
         <% some_filter = True %>
     %endif
     ## Patient
     %if task_filter.surname:
-        ${_("Surname")} = <b>${ task_filter.surname | h }</b>.
+        ${ _("Surname") } = <b>${ task_filter.surnameh }</b>.
         <% some_filter = True %>
     %endif
     %if task_filter.forename:
-        ${_("Forename")} = <b>${ task_filter.forename | h }</b>.
+        ${ _("Forename") } = <b>${ task_filter.forename }</b>.
         <% some_filter = True %>
     %endif
     %if task_filter.dob:
-        ${_("DOB")} = <b>${ format_datetime(task_filter.dob, DateFormat.SHORT_DATE) }</b>.
+        ${ _("DOB") } = <b>${ format_datetime(task_filter.dob, DateFormat.SHORT_DATE) }</b>.
         <% some_filter = True %>
     %endif
     %if task_filter.sex:
-        ${_("Sex")} = <b>${ task_filter.sex | h }</b>.
+        ${ _("Sex") } = <b>${ task_filter.sex }</b>.
         <% some_filter = True %>
     %endif
     %if task_filter.idnum_criteria:
-        ${_("ID numbers match one of:")}
+        ${ _("ID numbers match one of:") }
         ${ ("; ".join("{which} = <b>{value}</b>".format(
-                which=request.get_id_shortdesc(iddef.which_idnum),
-                value=iddef.idnum_value,
-            ) for iddef in task_filter.idnum_criteria) + ".") | h }
+                which=escape(request.get_id_shortdesc(iddef.which_idnum)),
```

Comment:
I think the current code is:

    %if task_filter.idnum_criteria:
        ${ _("ID numbers match one of:") }
        ${ ("; ".join("{which} = <b>{value}</b>".format(
                which=escape(request.get_id_shortdesc(iddef.which_idnum)),
                value=escape(iddef.idnum_value),
            ) for iddef in task_filter.idnum_criteria) + ".") | n }
        <% some_filter = True %>
    %endif

The description (``get_id_shortdesc``) is not intrinsically escaped, so needs escaping, as it is here. The value doesn't need escaping (it's an integer or, in principle, ``None`` -- it's part of ``IdNumReference``), so that ``escape()`` call is perhaps redundant. Then the whole thing has no extra Mako filters applied (``| n``). So could leave as it is or remove the ``escape()`` around ``iddef.idnum_value``. There are some checks to ensure that only integers get into this structure, but I guess no harm in continuing to escape it, which is the more conservative thing.